aea/test_tools/click_testing.py

```diff
@@ -28,19 +28,27 @@
 Links:
 
     https://github.com/pallets/click/issues/824
+    https://github.com/valory-xyz/open-aea/issues/171
+    https://github.com/valory-xyz/open-aea/issues/353
 
 """
+
 import shlex
 import sys
-from typing import Optional
+from contextlib import nullcontext, redirect_stderr
+from typing import ContextManager, Optional, cast
 
+from _pytest.capture import CaptureFixture  # type: ignore
 from click.testing import CliRunner as ClickCliRunner
 from click.testing import Result
 
 
 class CliRunner(ClickCliRunner):
     """Patch of click.testing.CliRunner."""
 
+    # NOTE: always set on the instance, never the class
```

Comment:
Why?

Reply:
we should always opt to change state on the instance instead of the class, because otherwise it will introduce a lasting change that can and will affect other parts of the code. If another test wants to use the `click.CliRunner.isolation` context manager, a test setting this at the class level can cause interference.